plugin/vscode/performance-writer.js

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2020  Matthew Glazar
+// See end of file for extended copyright information.
+
+const fs = require('fs');
+const path = require("path");
+const process = require("process");
+
+class PerformanceWriter {
+    
+    constructor (extensionContext) {
+        this._performanceTxtPath = path.join(process.cwd(), "test_performance.txt");
+        this._dataAttributes = {
+            columns:'timestamp,file-ref,file-name-or-duration,event-id'
+        };
+        this._fileName;
+        this._fileIndex;
+        this._extensionContext = extensionContext;
+    }
+
+    fileNameOptimizer (){
+        let currentSessionMap = this._extensionContext.globalState.get('qljsFileMap');
+        this._fileName = this._fileName.split(path.sep);
+        this._fileName = this._fileName[this._fileName.length - 1];
+        if (!(currentSessionMap)) {
+            currentSessionMap = {};
+            currentSessionMap['latestIndex'] = 1;
+            currentSessionMap[this._fileName] = 1;
+            this._fileIndex = 1;
+            this.writeToFile(null,"header");
+            this.writeToFile(null, "newFile");
+            this._extensionContext.globalState.update('qljsFileMap', currentSessionMap);
+        } else {
+            let existingFileIndex = currentSessionMap[this._fileName];
+            if (existingFileIndex) {
```

Comment:
Minor: This check will erroneously report that a file called `hasOwnProperty` or `toString`. Use a `Map` or do this instead:
```suggestion
            if (Object.prototype.hasOwnProperty.call(currentSessionMap, this._fileName)) {
                let existingFileIndex = currentSessionMap[this._fileName];
```

Reply:
followed the suggested change in latest commit